### ISSUE TITLE
ES5 vs ES6 vs TypeScript - The missing link

### ISSUE THREAD

**Issue by janrembold** [biotope/biotope-build#76]
Currently we have a mixed JavaScript architecture between "legacy" ES5 and "modern" ES6 TypeScript Plugins. TS Plugins are bundled with Webpack that has some painful performance problems. Also we don't support plain ES6 JavaScript Plugins. 

Problem is that I often see developers struggling with TypeScript when it is combined with external plugins that are not or faulty support typings. That makes the development process slow and painful. So I would really like to support ES6 Plugins as well.

First step is to remove webpack-stream plugin from gulp tasks and replace it with browserify/tsify... (https://github.com/biotope/biotope-build/issues/48) If this boosts performance and doesn't cause any memory leaks (like Webpack does currently) we can simply change the bundler silently.

Second step would be to support ES6 plugins that might be transformed to ES5 by e.g. Babelify. I propose to add a naming convention like "[pluginName].es6.js" to identify ES6 Plugins.

What do you think?

**Comment by SheepFromHeaven** [comment]
@janrembold I think you are mixing issues here, as we simply can support ES6 by adding the .js extension to the webpack config. We then still will have the performance issues, but we support ES6.
I would not combine the concerns here.

**Comment by janrembold** [comment]
@SheepFromHeaven You are right, this problem was mostly motivated by the performance issue. Not the feature itself. Closing this for now
